sequencer/src/state.rs

```diff
@@ -116,10 +114,31 @@ impl HotShotState for ValidatedState {
     /// proposal descends from parent. Returns updated `ValidatedState`.
     fn validate_and_apply_header(
         &self,
-        _instance: &Self::Instance,
+        instance: &Self::Instance,
         parent_header: &Self::BlockHeader,
         proposed_header: &Self::BlockHeader,
     ) -> Result<Self, Self::Error> {
+        // check header signagure
+        // in the signed message these are set to `None`.
+        let mut verifiable_header = proposed_header.clone();
+        let builder_signature = verifiable_header.builder_signature.take();
+        let _builder_address = verifiable_header.builder_address.take();
+        let _builder_fee_amount = verifiable_header.builder_fee_amount.take();
```

Comment:
This shouldn't be necessary, since the presence of `signature` should not affect the commitment scheme, and the address and amount actually should be included in the signed message (see earlier comment)